### PR TITLE
feat: Added seedable randomization

### LINKED ISSUE #10: Provide seedable randomization wrappers

### PR DESCRIPTION
**Pull request by AmenAnthem**
We still need to add documentation!

### CODE REVIEW

**Comment by AmenAnthem** [comment]
Resolves #10

File: internal/random/id.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package random
```
Comment (ChristoWolf) [inline internal/random/id.go:1]: Please document the package in one of the package's files.

File: internal/random/id.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	"errors"
+	"fmt"
+)
+
+// An Id represents the ID of an in-game entity. Its value is the integer representation of its original hex value.
+type Id struct {
```
Comment (ChristoWolf) [inline internal/random/id.go]: Is there a reason you prefer this struct over e.g. `type Id uint16`?

File: internal/random/id_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package random_test
+
+import (
+	"github.com/christowolf/vagrant-story-randomizer/internal/random"
+	"math"
+	"reflect"
+	"testing"
+)
+
+func TestNewId(t *testing.T) {
```
Comment (ChristoWolf) [inline internal/random/id_test.go:11]: Please document all test functions.

File: internal/random/randomizer.go [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package random
```
Comment (ChristoWolf) [inline internal/random/randomizer.go:2]: Regarding the commont from before: Maybe the package doc would best fit here.

File: internal/random/seed.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+func SeedFrom(source string) Seed {
+	return newSeed(int64(crc64.Checksum([]byte(source), crc64.MakeTable(crc64.ISO))))
+}
+
+func newSeed(source int64) Seed {
+	return Seed{source, rand.New(rand.NewSource(source))}
+}
+
+// Rand returns the rand.Rand of this Seed.
+func (s Seed) Rand() *rand.Rand {
```
Comment (ChristoWolf) [inline internal/random/seed.go]: Is this getter still needed?
I think itis only used inside the package, right?

File: internal/random/seed_test.go [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/christowolf/vagrant-story-randomizer/internal/random"
+	"reflect"
+	"testing"
+)
+
+func TestNewSeed(t *testing.T) {
+	t.Parallel()
+	t.Run("created seed is valid", func(t *testing.T) {
+		t.Parallel()
+		if got := random.NewSeed(); reflect.DeepEqual(got, random.Seed{}) {
```
Comment (ChristoWolf) [inline internal/random/seed_test.go:15]: They might still be equal with probability of about 1/512, right?

Comment (ChristoWolf) [reply internal/random/seed_test.go:15]: My bad, your test is of course correct.

**Review by ChristoWolf — CHANGES_REQUESTED** [review]
Looks great, only a few minor changes would be nice.

File: internal/random/id.go (hunk 2 above)
Comment (AmenAnthem) [reply internal/random/id.go]: I'm just used to writing wrappers, but as it's only a value the struct is unneccessary. I'm going to change that.

File: internal/random/seed.go (hunk 5 above)
Comment (AmenAnthem) [reply internal/random/seed.go]: Going to remove it, as it is currently not needed.

File: internal/random/id_test.go (hunk 3 above)
Comment (AmenAnthem) [reply internal/random/id_test.go:11]: Will do.

**Review by ChristoWolf — APPROVED** [review]
Looks good, thanks!

**Pull request merged**